### PR TITLE
feat(graphql/ml): Add custom properties to ml entities

### PR DESCRIPTION
**Pull request by asikowitz**
Tried to standardize the code here.

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/mlmodel/mappers/MLFeatureMapper.java [hunk 1]
```diff
@@ -138,10 +139,12 @@ private static void mapMLFeatureKey(@Nonnull MLFeature mlFeature, @Nonnull DataM
   private static void mapMLFeatureProperties(
       @Nullable final QueryContext context,
       @Nonnull MLFeature mlFeature,
-      @Nonnull DataMap dataMap) {
+      @Nonnull DataMap dataMap,
+      @Nonnull Urn entityUrn) {
     MLFeatureProperties featureProperties = new MLFeatureProperties(dataMap);
-    mlFeature.setFeatureProperties(MLFeaturePropertiesMapper.map(context, featureProperties));
-    mlFeature.setProperties(MLFeaturePropertiesMapper.map(context, featureProperties));
+    var graphqlProperties = MLFeaturePropertiesMapper.map(context, featureProperties, entityUrn);
```
Comment (chriscollins3456) [inline datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/mlmodel/mappers/MLFeatureMapper.java:145]: `var`? let's use the actual type here - same with other files below.

or we can just keep in inline like it was before

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/mappers/EmbeddedModelMapper.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+import com.linkedin.common.urn.Urn;
+import com.linkedin.datahub.graphql.QueryContext;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+/** Made for models that are embedded in other models and thus do not encode their own URN. */
+public interface EmbeddedModelMapper<I, O> {
+  O apply(
+      @Nullable final QueryContext context, @Nonnull final I input, @Nonnull final Urn entityUrn);
```
Comment (chriscollins3456) [inline datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/mappers/EmbeddedModelMapper.java:11]: i think this naming makes sense - not every embedded model mapper needs the entity urn, so not everything will need to implement this interface if it's not a top level model mapper obviously

**Review by chriscollins3456 — COMMENTED** [review]
looking good - main change is the `var` comment

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/mlmodel/mappers/MLFeatureMapper.java (hunk 1 above)
Comment (asikowitz) [reply datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/mlmodel/mappers/MLFeatureMapper.java:145]: Are we against using `var`? It's one of those cases where we have the same name for the model class and graphql class so imo it's less readable than just using `var`

**Review by chriscollins3456 — APPROVED** [review]

**Pull request merged**